### ISSUE TITLE
Some chart functions run out of colors (likely array index issue)

### ISSUE THREAD

**Issue by schanzer** [brownplt/code.pyret.org#507]
_I think this issue exists in most of our chart functions, but some are more resilient than others._

In [this starter file](https://code.pyret.org/editor#share=1lEhe_M_VU7ufyFynhmUwksYEf0WEscTf&v=0c33ff0), we import a large dataset. Try evaluating `stacked-bar-chart(shark-attacks-table, "ATTACK-TYPE", "COUNTRY")`, and you'll see that we quickly run out of colors for all the countries....and default to black. 

I believe one example of the bug can be traced to [this line in bar charts](https://github.com/brownplt/code.pyret.org/blob/horizon/src/web/js/trove/chart-lib.js#L790), where we naively try to get a slice of the `color_list` object using the number of legends as our slice length. $10 says that when we access the color at index `reallyBigNumber`, the resulting `undefined` gets turned into a pyret color with zeros for all the fields. In fact, this will likely happen at `notSoBigANumber`, because [the default color list](https://github.com/brownplt/code.pyret.org/blob/horizon/src/web/arr/trove/chart.arr#L1089) is pretty short!

I think the general solution here is that all charts should cycle through their color lists if they need more colors than the list allows. Bonus points if they start tweaking the colors programmatically. I would love to send a PR for this myself, but I'm overloaded with curriculum and PD staffing right now. :( 

(Could potentially be a great target for @asolove , if he's got an afternoon to kick around... 😉  )

**Comment by shriram** [comment]
Perhaps this is of use, since it's in the codebase already…

https://blog.brownplt.org/2018/06/11/philogenic-colors.html

**Comment by schanzer** [comment]
That's a great blog post. Two nits:

1. It's "phillotactic", not "philogenic", right?
2. There's no documentation of how a developer might _use_ this infrastructure. If it's not listed in the API, can you give us a pointer to where it is in the codebase?

**Comment by shriram** [comment]
Sorry, not sure. No real time to find it. I'd say look for the place in the
code where error messages are generated, it's got to be referenced from
there.

**Comment by asolove** [comment]
I'm looking into this.

For reference: the color generation code used for error highlighting, under discussion above, is here: https://github.com/brownplt/code.pyret.org/blob/1701bc61e67e04342cb9d68c30a6a70d552aa50e/src/web/js/output-ui.js#L47

**Comment by asolove** [comment]
@schanzer I can reproduce this in production, but am not able to get a similar `stacked-bar-chart` locally. Can you share with me (by email if sensitive) the content that's in the context used in that file, so I can call `stacked-bar-chart` locally?

**Comment by schanzer** [comment]
@asolove of course! [The entire context can be found here](https://code.pyret.org/editor#share=1l2A-dbab4Plb1kYErq16kktzfsQ_suMv&v=0c33ff0), but what matters is the definition of `stacked-bar-chart` on line 441. 

Thank you for looking into this! 🙏

**Comment by schanzer** [comment]
If the number of requested colors is greater than the length of the color palette, use Pyret's internal algorithm
